### PR TITLE
Generate device plugin LinkType selector from NodePolicy LinkType

### PR DESCRIPTION
**Pull request by Mmduh-483**

### CODE REVIEW

File: controllers/sriovnetworknodepolicy_controller.go [hunk 1]
```diff
@@ -553,6 +554,13 @@ func (r *SriovNetworkNodePolicyReconciler) renderDevicePluginConfigData(pl *srio
 			if len(p.Spec.NicSelector.PfNames) > 0 {
 				netDeviceSelectors.PfNames = sriovnetworkv1.UniqueAppend(netDeviceSelectors.PfNames, p.Spec.NicSelector.PfNames...)
 			}
+			if p.Spec.LinkType != "" && !sriovnetworkv1.StringInArray(p.Spec.LinkType, netDeviceSelectors.LinkTypes) {
```
Comment (moshe010) [inline controllers/sriovnetworknodepolicy_controller.go:557]: in what case with p.Spec.LinkType == ""?

Comment (moshe010) [inline controllers/sriovnetworknodepolicy_controller.go:557]: !sriovnetworkv1.StringInArray(p.Spec.LinkType, netDeviceSelectors.LinkTypes) this is to check if that linktype selector already exist?

File: controllers/sriovnetworknodepolicy_controller.go [hunk 2]
```diff
@@ -553,6 +554,13 @@ func (r *SriovNetworkNodePolicyReconciler) renderDevicePluginConfigData(pl *srio
...
+				linkType := linkTypeEthernet
+				if strings.ToLower(p.Spec.LinkType) == "ib" {
+					linkType = linkTypeInfiniband
+				}
+				netDeviceSelectors.LinkTypes = append(netDeviceSelectors.LinkTypes, linkType)
```
Comment (zshi-redhat) [inline controllers/sriovnetworknodepolicy_controller.go:562]: Might be good to us sriovnetworkv1.UniqueAppend?

File: controllers/sriovnetworknodepolicy_controller.go (hunk 1 above)
Comment (zshi-redhat) [inline controllers/sriovnetworknodepolicy_controller.go:557]: netDeviceSelectors.LinkTypes use different value than p.Spec.LinkType, for example:
the former may includes `infiniband`, the latter may be `ib`. so the conditional check here would always true if p.Spec.LinkType != "" ?

Comment (zshi-redhat) [reply controllers/sriovnetworknodepolicy_controller.go:557]: This API filed is omitempty.

**Pull request merged**